src/WebJobs.Extensions.DurableTask/Correlation/DurableTaskCorrelationTelemetryInitializer.cs

```diff
@@ -0,0 +1,318 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using DurableTask.Core;
+using DurableTask.Core.Settings;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Correlation
+{
+    /// <summary>
+    /// Telemetry Initializer that sets correlation ids for W3C.
+    /// This source is based on W3COperationCorrelationTelemetryInitializer.cs
+    /// 1. Modified with CorrelationTraceContext.Current
+    /// 2. Avoid to be overriden when it is RequestTelemetry
+    /// Original Source is here https://github.com/microsoft/ApplicationInsights-dotnet-server/blob/2.8.0/Src/Common/W3C/W3COperationCorrelationTelemetryInitializer.cs.
+    /// </summary>
+    [Obsolete("Not ready for public consumption.")]
```

Comment:
Is there a reason we are marking this as obsolete? In general, since this is internal, I don't see much benefit to this.

Reply:
I think you may want to mark the interface `ITelemetryInitializer` with this attribute, as that _is_ public. I wish it didn't have to be, but I haven't found a good way to have non-public interfaces be injected via dependency injection yet.

Reply:
I remove the Obsolute attribute. The reason it was there is, I took this code from https://github.com/microsoft/ApplicationInsights-dotnet-server/blob/2.8.0/Src/Common/W3C/W3COperationCorrelationTelemetryInitializer.cs as I leave the comment on it. After that I modify it. So the first time, I just keep the same structure, however, since I modify it, it means nothing. I removed it and remove warning suppression at the TelemetryActivator.